### PR TITLE
Update to use Jupyterlab 3.3.0rc0

### PR DESCRIPTION
**Pull request by akchinSTC**
Have to straight pin to 3.3.0rc0, otherwise will pull in alpha builds of 4.x

### How was this pull request tested?

 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Comment by akchinSTC** [comment]
Can one of the admins verify this patch?

File: setup.py [hunk 1]
```diff
@@ -80,7 +80,7 @@
         'jupyter_core>=4.0,<5.0',
         'jupyter_client>=6.1.7',
         'jupyter_server>=1.7.0',
-        'jupyterlab>=3.0.17',
+        'jupyterlab==3.3.0rc0',
```
Comment (ptitzler) [inline setup.py:83]: Did you try 

```suggestion
        'jupyterlab~=3.3.0rc0',
```

If I'm not mistaken this should pick up any release candidate

**Review by ptitzler — APPROVED** [review]
LGTM. Created a pipeline from scratch and ran it on KFP.

**Review by lresende — CHANGES_REQUESTED** [review]
We also need to update the package.json files.

File: package.json [hunk 2]
```diff
@@ -41,13 +41,13 @@
...
     "@cypress/webpack-preprocessor": "^5.5.0",
-    "@jupyterlab/testutils": "3.0.0",
+    "@jupyterlab/testutils": "3.0.0-rc.0",
     "@testing-library/cypress": "^7.0.4",
     "@types/jest": "^26.0.20",
     "@types/lodash": "^4.14.170",
     "@types/node": "^15.0.1",
     "@types/react": "^17.0.0",
-    "@types/react-dom": "^17.0.0",
+    "@types/react-dom": "~17.0.0",
```
Comment (lresende) [inline package.json:50]: I would leave as `^` which is what Lab uses.

File: packages/metadata-common/package.json [hunk 3]
```diff
@@ -35,10 +35,10 @@
...
-    "@jupyterlab/apputils": "^3.0.0",
-    "@jupyterlab/codeeditor": "^3.0.0",
-    "@jupyterlab/ui-components": "^3.0.0",
+    "@jupyterlab/application": "^3.3.0-rc.0",
+    "@jupyterlab/apputils": "^3.3.0-rc.0",
+    "@jupyterlab/codeeditor": "^3.3.0-rc.0",
+    "@jupyterlab/ui-components": "^3.3.0-rc.0",
     "@lumino/algorithm": "^1.3.3",
     "@lumino/messaging": "^1.4.3",
     "@lumino/widgets": "^1.17.0",
```
Comment (lresende) [inline packages/metadata-common/package.json:44]: Widgets needs to bum to 1.19
https://github.com/jupyterlab/jupyterlab/blob/b71275f8b67a662441eacfba3def4312beeae1f4/packages/fileeditor/package.json#L52

File: packages/r-editor/package.json [hunk 4]
```diff
@@ -36,15 +36,15 @@
...
+    "@jupyterlab/application": "^3.3.0-rc.0",
+    "@jupyterlab/apputils": "^3.3.0-rc.0",
+    "@jupyterlab/codeeditor": "^3.3.0-rc.0",
+    "@jupyterlab/docregistry": "^3.3.0-rc.0",
+    "@jupyterlab/filebrowser": "^3.3.0-rc.0",
+    "@jupyterlab/fileeditor": "^3.3.0-rc.0",
+    "@jupyterlab/launcher": "^3.3.0-rc.0",
+    "@jupyterlab/mainmenu": "^3.3.0-rc.0",
+    "@jupyterlab/settingregistry": "^3.3.0-rc.0",
     "@lumino/coreutils": "^1.5.3"
```
Comment (lresende) [inline packages/r-editor/package.json:48]: coreutils needs to go to 1.5.6 to fix https://github.com/elyra-ai/elyra/security/dependabot/45

**Review by lresende — CHANGES_REQUESTED** [review]
Good progress :) but unfortunately, we will also need to check the lab 3.3.0 branch and see what they are using for the lumino versions and we should match. Yes, it's a tedious manual process.

File: packages/metadata-common/package.json (hunk 3 above)
Comment (akchinSTC) [reply packages/metadata-common/package.json]: widgets is still resolving to 1.17 via yarn.lock. need to check what dep is resolving to that

Comment (lresende) [reply packages/metadata-common/package.json]: I believe you might need to do a yarn or yarn update from root to update lock?

**Comment by lresende** [comment]
> widgets is still resolving to 1.17 via yarn.lock.
Please update the version on package.json in the root.

**Comment by akchinSTC** [comment]
> Good progress :) but unfortunately, we will also need to check the lab 3.3.0 branch and see what they are using for the lumino versions and we should match. Yes, it's a tedious manual process.

Should we ease some of these forced resolutions on node packages?

**Comment by akchinSTC** [comment]
currently failing with the following output : 
```
@elyra/services: $ jest
@elyra/services: ts-jest[versions] (WARN) Version 4.1.6 of typescript installed has not been tested with ts-jest. If you're experiencing issues, consider using a supported version (>=2.7.0 <4.0.0). Please do not report issues in ts-jest if you are using unsupported versions.
@elyra/services: FAIL src/test/application.spec.ts
@elyra/services:   ● Test suite failed to run
@elyra/services:     /Users/akchin/github/elyra/node_modules/@jupyterlab/codemirror/lib/index.js:7
@elyra/services:     import { CodeMirrorEditorFactory } from './factory';
@elyra/services:     ^^^^^^
@elyra/services:     SyntaxError: Cannot use import statement outside a module
@elyra/services:       at ScriptTransformer._transformAndBuildScript (../../node_modules/@jest/transform/build/ScriptTransformer.js:537:17)
@elyra/services:       at ScriptTransformer.transform (../../node_modules/@jest/transform/build/ScriptTransformer.js:579:25)
@elyra/services:       at Object.<anonymous> (../../node_modules/@jupyterlab/testutils/src/notebook-utils.ts:6:1)
@elyra/services: Test Suites: 1 failed, 1 total
@elyra/services: Tests:       0 total
@elyra/services: Snapshots:   0 total
@elyra/services: Time:        2.583s
@elyra/services: Ran all test suites.
@elyra/services: error Command failed with exit code 1.
@elyra/services: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Review by lresende — APPROVED** [review]
LGTM, are we waiting for the final release on the 28th? Or is this ready to merge?

**Pull request merged**